src/+otp/+pendulumdae/PendulumDAEProblem.m

```diff
@@ -0,0 +1,54 @@
+classdef PendulumDAEProblem < otp.Problem
+    %CONSTRAINEDPENDULUM PROBLEM This is a Hessenberg Index-2 DAE problem posed in terms of three constraints
+    %
+    
+    properties (SetAccess=protected)
+        RHSDifferential
+        RHSAlgebraic
+    end
+
+    properties (Dependent)
+      Y0Differential
+      Z0Algebraic
+    end
+
+    methods
+        function obj = PendulumDAEProblem(timeSpan, y0, parameters)
+            obj@otp.Problem('Constrained Pendulum', 7, timeSpan, y0, parameters);
+        end
+    end
+    
+    methods (Access = protected)
+        function onSettingsChanged(obj)
+            m = obj.Parameters.Mass;
+            l = obj.Parameters.Length;
+            g = obj.Parameters.Gravity;
+
+            % get initial energy
+            initialconstraints = otp.pendulumdae.constraints([], obj.Y0Differential, g, m, l, 0);
+            E0 = initialconstraints(3);
+            
+
+            % The right hand size in terms of x, y, x', y', and three control parameters z
+            obj.RHS = otp.RHS(@(t, y) otp.pendulumdae.f(t, y, g, m, l, E0), ...
+                'Mass', @(t, y) otp.pendulumdae.mass(t, y, g, m, l, E0));
```

Comment:
Since the mass matrix is constant, it can be a matrix instead of a function handle.  See https://github.com/ComputationalScienceLaboratory/ODE-Test-Problems/blob/3d4cb2262931eaf261ca3061479b81928b1ef419/src/%2Botp/%2Bzlakinetics/ZlaKineticsProblem.m#L19 for example

Reply:
I was just copying the pendulum problem, which had @(t,y). I will change